packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,702 @@
+# Integrating a custom plugin in the CLI
+
+One of the main features of Code PushUp is the ability to write custom plugins and track your own metrics.
+It enables you to implement nearly any kind of metric you want to track progress with minimum effort.
+In this section we will go through the steps needed to create a custom plugin and integrate it in your project.
+
+## Setup the core config
+
+To start crafting custom plugins you need a minimum `code-pushup.config.(ts|js|mjs)` file maintaining a `plugins` property.
+property.
+All plugins are registered in a core config object and can take potential options to configure its behaviour.
+The following example shows where to register the plugin:
+
+```typescript
+// code-pushup.config.ts
+import { create as myPlugin } from 'my-plugin';
+
+export default {
+  plugins: [
+    // can return `PluginConfig` or `Promise<PluginConfig>`
+    await myPlugin({
+      // plugin options here
+    }),
+  ],
+};
+```
+
+## Plugin Structure
+
+Every plugin is defined over a [`PluginConfig`](@TODO).
+
+The plugin config maintains:
+
+- metadata about the plugin [PluginMeta](@TODO)
+- metadata about the available audits [`Audit`](@TODO)
+- internal logic that produces the plugin output as [`AuditOutputs`](@TODO).
```

Comment:
I'd introduce the word "runner" here, so that the user can start catching on to our terminology.